src/panels/lovelace/card-features/hui-fan-oscillate-card-feature.ts

```diff
@@ -0,0 +1,157 @@
+import type { PropertyValues, TemplateResult } from "lit";
+import { LitElement, html } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { styleMap } from "lit/directives/style-map";
+import { computeDomain } from "../../../common/entity/compute_domain";
+import { stateColorCss } from "../../../common/entity/state_color";
+import "../../../components/ha-control-select";
+import type { ControlSelectOption } from "../../../components/ha-control-select";
+import { UNAVAILABLE } from "../../../data/entity";
+import type { FanEntity } from "../../../data/fan";
+import { FanEntityFeature } from "../../../data/fan";
+import type { HomeAssistant } from "../../../types";
+import type { LovelaceCardFeature } from "../types";
+import { cardFeatureStyles } from "./common/card-feature-styles";
+import type {
+  FanOscillateCardFeatureConfig,
+  LovelaceCardFeatureContext,
+} from "./types";
+import { supportsFeature } from "../../../common/entity/supports-feature";
+import type { LocalizeKeys } from "../../../common/translations/localize";
+
+export const supportsFanOscilatteCardFeature = (
+  hass: HomeAssistant,
+  context: LovelaceCardFeatureContext
+) => {
+  const stateObj = context.entity_id
+    ? hass.states[context.entity_id]
+    : undefined;
+  if (!stateObj) return false;
+  const domain = computeDomain(stateObj.entity_id);
+  return (
+    domain === "fan" && supportsFeature(stateObj, FanEntityFeature.OSCILLATE)
+  );
+};
+
+@customElement("hui-fan-oscillate-card-feature")
+class HuiFanOscillateCardFeature
+  extends LitElement
+  implements LovelaceCardFeature
+{
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @property({ attribute: false }) public context?: LovelaceCardFeatureContext;
+
+  @state() private _config?: FanOscillateCardFeatureConfig;
+
+  @state() _oscillate?: boolean;
+
+  private get _stateObj() {
+    if (!this.hass || !this.context || !this.context.entity_id) {
+      return undefined;
+    }
+    return this.hass.states[this.context.entity_id!] as FanEntity | undefined;
+  }
+
+  static getStubConfig(): FanOscillateCardFeatureConfig {
+    return {
+      type: "fan-oscillate",
+    };
+  }
+
+  public setConfig(config: FanOscillateCardFeatureConfig): void {
+    if (!config) {
+      throw new Error("Invalid configuration");
+    }
+    this._config = config;
+  }
+
+  protected willUpdate(changedProp: PropertyValues): void {
+    if (
+      (changedProp.has("hass") || changedProp.has("context")) &&
+      this._stateObj
+    ) {
+      const oldHass = changedProp.get("hass") as HomeAssistant | undefined;
+      const oldStateObj = oldHass?.states[this.context!.entity_id!];
+      if (oldStateObj !== this._stateObj) {
+        this._oscillate = this._stateObj.attributes.oscillating as boolean;
+      }
+    }
+  }
+
+  private async _valueChanged(ev: CustomEvent) {
+    const shouldOscillate = (ev.detail as any).value as boolean;
```

Comment:
`ev.detail.value` is `"yes" | "no"` as that is what you passed to `ha-control-select` but you are treating it as a boolean.